lib/src/klio/transforms/decorators.py

```diff
@@ -49,6 +49,40 @@
 )
 
 
+class ThreadLimitGenerator(object):
```

Comment:
Optional thought: would tucking this class in with the `ThreadLimiter` class et al be a better fit? I could see a reason for either here or there so your call.

Reply:
Yeah I debated that too.  I ultimately put it in `decorators.py` because at least for now that's all that uses it and I don't immediately foresee any other potential use.  If we ever find another use or a need to generalize it for reusability then we'd probably want to move it.